06-building-a-program-start-to-finish/writing-tests/output/InvertedIndexSpec.js

```diff
@@ -0,0 +1,345 @@
+        .toBeTruthy();
+    });
+  });
+
+  describe('The createIndex method', () => {
+    const book = {
+      0: {
+        title: 'my story',
+        text: "who does'nt wish for a lovely story",
+      },
+      1: {
+        title: 'my wishes',
+        text: 'story of a lovely unicorn',
+      },
+    };
+    let result = '';
+    beforeEach((done) => {
+      iDex.createIndex('mybook', book)
+        .then((response) => {
+          result = response;
+          done();
+        });
+    });
+    it('should return an inverted index', () => {
+      const expected = {
+        my: [0, 1],
+        story: [0, 1],
+        who: [0],
+        doesnt: [0],
+        wish: [0],
+        for: [0],
+        a: [0, 1],
+        lovely: [0, 1],
+        wishes: [1],
+        of: [1],
+        unicorn: [1],
+      };
+      expect(result).toEqual(expected);
+    });
+  });
+
+  describe('The getIndex method', () => {
+    it('should be able to retreive all indexes of a book already indexed',
+      () => {
+        expect(iDex.getIndex('books')).toEqual({
+          how: [0],
+          to: [0],
+          read: [0],
+          a: [0],
+          book: [0],
+          men: [0],
+          are: [0],
+          rational: [0],
+          animals: [0, 1],
+          their: [0],
+          rationality: [0],
+          agreement: [0],
+          is: [0],
+          the: [0, 1],
+          source: [0],
+          of: [0, 1],
+          power: [0],
+          agree: [0],
+          naked: [1],
+          ape: [1],
+          indeed: [1],
+          we: [1],
+          have: [1],
+          most: [1],
+          subtle: [1],
+          and: [1],
+          complex: [1],
+          facial: [1],
+          expression: [1],
+          system: [1],
+          all: [1],
+          living: [1] });
+      });
+
+    it('should be able to retreive all indexes ' +
+      ' of a word in a book already indexed', () => {
+
+      expect(iDex.getIndex('books').complex).toEqual([1]);
+      expect(iDex.getIndex('books').rationality).toEqual([0]);
+      expect(iDex.getIndex('books').facial).toEqual([1]);
+    });
+
+    it('should return an undefined for unavailable words', () => {
+      expect(iDex.getIndex('books').africa).toBeUndefined();
+      expect(iDex.getIndex('books').andela).toBeUndefined();
+      expect(iDex.getIndex('books').tdd).toBeUndefined();
+    });
+  });
+
+  describe('The searchIndex method', () => {
+    it('should an empty result if no token was searched for', () => {
+      expect(iDex.searchIndex('books', '')).toEqual([{
+        books: {
+          '': [],
+        },
+      }]);
+    });
+
+    it('should return apropriate result', () => {
+      expect(iDex.searchIndex('books', 'complex read animals')).toEqual([{
+        books: {
+          complex: [1],
+          read: [0],
+          animals: [0, 1],
+        },
+      }]);
+    });
+
+    it('should return apropriate result when ' +
```

Comment:
apropriate or appropriate?  - Please correct this typo here and also where it reoccurs